### PR TITLE
refactor(bestax-migrate): extract resolveBooleanish and sweep the six booleanish-literal sites

### LINKED ISSUE #563: refactor(bestax-migrate): extract resolveBooleanish and sweep the six booleanish-literal sites
**Goal:** introduce `resolveBooleanish(attr)` in `jsx-utils.ts` returning `truthy | falsy | expression` (resolving *every* statically-known literal by truthiness, matching RBC's runtime `!!value`), then rewire all six sites atomically **under one test** so the classification lives in exactly one place.

| # | Site | File / line | Current shape | Notes |
|---|------|-------------|---------------|-------|
| 1 | `button` `remove` | `specials.ts:214` | `literal.kind !== 'expression'` | The #558 convention — the intended target shape. |
| 2 | `heading` | `specials.ts:251` | `kind !== 'expression' && !!value` | Landed in #559. |
| 3 | `subtitle` | `specials.ts` (subtitle handler) | same as heading | Landed in #559. |
| 4 | `booleanToProp` | `props.ts:111` | gates on `literal.kind === 'boolean'` | **Drift #4**: a truthy string literal (`foo="true"`) wrongly takes the dynamic-TODO branch instead of resolving. Aligning it fixes that contradiction as a side effect. |
| 5 | `panel-tab` / `breadcrumb-item` `active` | `specials.ts:707`, `:743` | older `active` ladders (`kind !== 'boolean'`) | Same classification, different downstream branching — only the classification moves. |
| 6 | `field` `multiline` | `specials.ts:469` / `:510` | **presence-only** (`findAttr` only; value never read) | **Presence-vs-value bug**: `multiline={false}` and `multiline={expr}` both silently become `grouped="multiline"`. Should resolve by value — `false` → drop, dynamic → keep + `prop:multiline` TODO. Needs a fixture pair + unit coverage, same as #559 did for `heading`. |

**Scope notes:**
- The sites share only the *classification*, not the downstream branching (`booleanToProp` carries a `value` payload plus a `value === undefined` pure-rename branch; the `active` ladders and `field` differ again) — so `resolveBooleanish` replaces the ~4 classification lines at each site, not the whole handler.
- Site 6 is a genuine behavioural fix, not just a refactor.
- One PR across all six under one test — a helper coexisting with inline copies is a worse drift surface than N consistent inline copies, so a partial extract is explicitly not wanted.
- Package hard rule holds: dynamic values keep the attr + get a `TODO(bestax-migrate)`, never a best-guess rewrite.

Refs #559, #558.

### PR DESCRIPTION
**Pull request by bestaxbot**
## Summary

Introduces `resolveBooleanish(attr)` in `jsx-utils.ts`, returning `'truthy' | 'falsy' | 'expression'` by resolving every statically-known literal by truthiness (matching RBC's runtime `!!value`), and rewires all six sites onto it so the classification lives in exactly one place:

1. Button `remove` (`specials.ts`) — drop-in classification swap.
2. Heading `heading` (`specials.ts`) — drop-in classification swap.
3. Heading `subtitle` (`specials.ts`) — drop-in classification swap.
4. `booleanToProp` (`props.ts`) — fixes drift #4: a truthy string literal (e.g. `rounded="true"`) previously fell into the dynamic-TODO branch instead of resolving; now it resolves like every other site.
5. `panel-tab` / `breadcrumb-item` `active` ladders (`specials.ts`) — same classification now shared; a static string/number literal (e.g. `active="true"`) resolves instead of being misclassified as dynamic.
6. `field` `multiline` (`specials.ts`) — the presence-vs-value bug: `multiline={false}` and `multiline={expr}` both used to silently become `grouped="multiline"`. Restructured per the issue's trap notes:
   - the attr-strip loop is now conditional — a dynamic `multiline` survives on the element instead of always being stripped
   - falsy `multiline` with no `kind`/`align` now emits nothing (previously fell through to a bare `grouped` attribute)
   - the addons "dropped from this addons Field" TODO only fires for the resolved-truthy case; falsy gets no TODO; dynamic gets a generic "resolve the `grouped`/`hasAddons` combination by hand" TODO
   - `ctx.dirty` is only set when something in the element actually changed
   - every return path now includes `handledProps: ['multiline']` — without it, the mapping's fallback `multiline: { drop: true }` prop action was silently stripping the dynamic attr right after the special handler decided to keep it (caught this via the new test fixture)

For the open question in the issue's site-6 notes (dynamic `multiline` on an addons Field), went with keeping the attr + a TODO — matching how the `heading` handler treats its own dynamic case — rather than dropping it with the incompatibility TODO.

## Test plan

- [x] New fixture pair `__testfixtures__/field-multiline-literals.{input,output}.tsx` covering the falsy/dynamic × group/addons matrix for site 6
- [x] Direct unit test for `resolveBooleanish` in `jsx-utils.test.ts`
- [x] New unit tests in `transform.test.ts` for the `booleanToProp` drift fix (site 4) and the `active`-ladder classification fix (site 5)
- [x] Confirmed all existing fixtures (`edge-cases-2`, `form`) are byte-identical — no churn
- [x] `pnpm --filter bestax-migrate run lint` — pass
- [x] `pnpm --filter bestax-migrate run typecheck` — pass
- [x] `pnpm --filter bestax-migrate run test:coverage` — pass (271 tests)
- [x] `pnpm run format:check` — pass
- [x] `pnpm run gen:catalog:check` — pass
- [x] `pnpm run check:conformance` — pass

Fixes #563

Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

**Review by allxsmith — APPROVED** [review]

File: bestax-migrate/src/sources/react-bulma-components/specials.ts [hunk 1]
```diff
@@ -481,33 +484,60 @@ const SPECIALS: Record<string, SpecialHandler> = {
         'prop:kind',
         'dynamic Field kind/align; map to the bestax `grouped` / `hasAddons` props by hand'
       );
-      return {};
+      // This handler owns `multiline` end-to-end; without this, the mapping's
+      // fallback `multiline: { drop: true }` prop action would silently strip
+      // a dynamic `multiline` the moment kind/align themselves are dynamic.
+      return { handledProps: ['multiline'] };
```
[the remark this answers is not in the document]
Comment (allxsmith) [reply bestax-migrate/src/sources/react-bulma-components/specials.ts:490]: @bestaxbot it looks like we did not address copilots findings.   Can you look at copilot's comments?   See if we should address these?****

[the remark this answers is not in the document]
Comment (bestaxbot) [reply bestax-migrate/src/sources/react-bulma-components/specials.ts:490]: This branch is the dynamic kind/align case, and I left it deliberate rather than tighten it. When kind/align is an expression we can't resolve the grouped/hasAddons combination at all, so the whole Field already carries a prop:kind 'map to grouped/hasAddons by hand' TODO covering the entire element — multiline included. Two consequences of that choice:

- Dynamic multiline: kept on the element with no separate prop:multiline TODO because the prop:kind TODO already tells the developer to resolve the whole combination by hand; a second comment on the same statement would just be noise (addTodo dedupes per rule anyway). This matches how the resolved-kind='addons' path treats multiline as part of one grouped/hasAddons decision.
- Falsy literal multiline (e.g. multiline={false}): yes, it survives as a dead prop here. It's harmless — Field's index signature absorbs it and it typechecks — and stripping it would silently drop information from an element we're explicitly punting to manual resolution, which cuts against the 'never silently lose info' rule. The handledProps: ['multiline'] is only there to stop the fallback drop action from clobbering a *dynamic* multiline the instant kind/align go dynamic.

I'd rather keep the honest 'this whole Field needs hand-resolution' signal than partially clean it. Happy to strip the falsy-literal case in this branch too if you'd prefer it tidier, but I don't think it's a defect as-is.
